scripts/ci/in_container/entrypoint_ci.sh

```diff
@@ -0,0 +1,220 @@
+    echo
+    echo "Using ${HADOOP_DISTRO:=} distribution of Hadoop from ${HADOOP_HOME:=}"
+    echo
+fi
+
+AIRFLOW_ROOT="$(cd ${MY_DIR}; cd ../../..; pwd)"
+
+export AIRFLOW__CORE__DAGS_FOLDER="${AIRFLOW_SOURCES}/tests/dags"
+
+# add test/test_utils to PYTHONPATH (TODO: Do we need it?)
+export PYTHONPATH=${PYTHONPATH:-${AIRFLOW_SOURCES}/tests/test_utils}
+
+export AIRFLOW__CORE__UNIT_TEST_MODE=True
+export HADOOP_DISTRO
+
+echo "#######################################################################"
+echo
+echo "  You can drop into ipdb debugger by adding this line to your code:"
+echo
+echo "         import ipdb; ipdb.set_trace()"
+echo
+echo "  Then run your tests with 'run-tests <TEST> --nocapture' "
+echo
+echo "#######################################################################"
+
+
+# Fix codecov build path
+# TODO: Check this - this should be made travis-independent
+if [[ ! -h /home/travis/build/apache/airflow ]]; then
+  sudo mkdir -p /home/travis/build/apache
+  sudo ln -s ${AIRFLOW_ROOT} /home/travis/build/apache/airflow
+fi
+
+# Fix file permissions
+if [[ -d $HOME/.minikube ]]; then
+    sudo chown -R airflow.airflow $HOME/.kube $HOME/.minikube
+fi
+
+# Cleanup the logs when entering the environment
+sudo rm -rf ${AIRFLOW_HOME}/logs/*
+
+if [[ "${ENV}" == "docker" ]]; then
+    # Start MiniCluster
+    java -cp "/tmp/minicluster-1.1-SNAPSHOT/*" com.ing.minicluster.MiniCluster >/dev/null 2>&1 &
+
+    # Set up ssh keys
+    echo 'yes' | ssh-keygen -t rsa -C your_email@youremail.com -P '' -f ~/.ssh/id_rsa >/dev/null 2>&1
+    cat ~/.ssh/id_rsa.pub >> ~/.ssh/authorized_keys
+    ln -s -f ~/.ssh/authorized_keys ~/.ssh/authorized_keys2
+    chmod 600 ~/.ssh/*
+
+    # SSH Service
+    sudo service ssh restart >/dev/null 2>&1
+
+    # Setting up kerberos
+
+    FQDN=`hostname`
+    ADMIN="admin"
+    PASS="airflow"
+    KRB5_KTNAME=/etc/airflow.keytab
+
+    if [[ ${AIRFLOW_CI_VERBOSE} == "true" ]]; then
+        echo
+        echo "Hosts:"
+        echo
+        cat /etc/hosts
+        echo
+        echo "Hostname: ${FQDN}"
+        echo
+    fi
+
+    sudo cp ${MY_DIR}/krb5/krb5.conf /etc/krb5.conf
+
+    echo -e "${PASS}\n${PASS}" | \
+        sudo kadmin -p ${ADMIN}/admin -w ${PASS} -q "addprinc -randkey airflow/${FQDN}" >/dev/null 2>&1
+    sudo kadmin -p ${ADMIN}/admin -w ${PASS} -q "ktadd -k ${KRB5_KTNAME} airflow" >/dev/null 2>&1
+    sudo kadmin -p ${ADMIN}/admin -w ${PASS} -q "ktadd -k ${KRB5_KTNAME} airflow/${FQDN}" >/dev/null 2>&1
+    sudo chmod 0644 ${KRB5_KTNAME}
+fi
+
+# Exporting XUNIT_FILE so that we can see summary of failed tests
+# at the end of the log
+export XUNIT_FILE=${AIRFLOW_HOME}/logs/all_tests.xml
+mkdir -pv ${AIRFLOW_HOME}/logs/
+
+cp -f ${MY_DIR}/airflow_ci.cfg ${AIRFLOW_HOME}/unittests.cfg
+
+# If we do not want to run tests, we simply drop into bash
+if [[ "${RUN_TESTS}" == "false" ]]; then
+    if [[ -z "${ARGS[*]}" ]]; then
+        exec /bin/bash
+    else
+        exec /bin/bash -c "${ARGS[*]}"
```

Comment:
```suggestion
        exec /bin/bash -c "$(printf "%q " "${ARGS[@]}")"
```

to preserve quotes/spaces/special chars in the args.

Reply:
OK. Did not know this trick. I indeed only recently (after running a number of shellchecks) found better way of escaping/Arrays - it's always tricky subject. I will make sure to review all cases for that.

Reply:
This is always the most difficult one to handle - preserving quoting/spaces/special chars to a single string/command.

Reply:
This tricks is part of oozie-to-airflow also:
https://github.com/GoogleCloudPlatform/oozie-to-airflow/blob/master/o2a/scripts/git.sh

Reply:
I think I nailed it !